### PR TITLE
Print instatneously using ipython_display

### PR DESCRIPTION
**Pull request by MohamedElKamhawy**

### CODE REVIEW

File: remotespark/livyclientlib/livysession.py [hunk 1]
```diff
@@ -76,10 +77,10 @@ def create_sql_context(self):
 
         self.logger.debug("Starting '{}' sql and hive session.".format(self.kind))
 
-        self.ipython_display.writeln("Creating SqlContext as 'sqlContext'")
+        self.ipython_display.writeln_and_flush("Creating SqlContext as 'sqlContext'")
         self._create_context(Constants.context_name_sql)
 
-        self.ipython_display.writeln("Creating HiveContext as 'hiveContext'")
+        self.ipython_display.writeln_and_flush("Creating HiveContext as 'hiveContext'")
```
Comment (aggFTW) [inline remotespark/livyclientlib/livysession.py]: I'm against this rename. Leave it as writeln

File: remotespark/utils/ipythondisplay.py [hunk 2]
```diff
@@ -18,5 +19,9 @@ def write(self, msg):
     def writeln(self, msg):
         self.write("{}\n".format(msg))
```
Comment (aggFTW) [inline remotespark/utils/ipythondisplay.py]: Flush should happen in write

File: remotespark/utils/ipythondisplay.py [hunk 3]
```diff
@@ -18,5 +19,9 @@ def write(self, msg):
...
 
+    def writeln_and_flush(self, msg):
```
Comment (aggFTW) [inline remotespark/utils/ipythondisplay.py]: Do not introduce a new function. Just use writeln

**Comment by aggFTW** [comment]
Did just test this? Is there a way we could unit test this?

**Comment by aggFTW** [comment]
What about other methods, like `send_error`?

**Comment by msftristew** [comment]
@aggFTW, I suggested that there be a new method for writing a message and flushing the output stream.  It is not generally necessary to flush the output stream after every write and that can have performance implications.

**Comment by aggFTW** [comment]
Class `ipythondisplay` is intended to be used as a way for us to show something to the user instantaneously. It's our view. It should always show immediately to the user.

**Comment by msftristew** [comment]
You can only unit test this by mocking `sys.stdout` -- possible, but nothing else in the class is mockable either.

**Comment by msftristew** [comment]
Obviously this is the only circumstance in which Jupyter's output buffering has ever been an issue for us.  I don't see why we would flush the output buffer after every write, but if we're confident there will never be a situation where it would ever trouble us from a performance standpoint, then I guess it's fine.

Note that while I haven't tested it, flushing `sys.stderr` will *probably* also work for errors.

**Comment by MohamedElKamhawy** [comment]
@aggFTW  does send_error have the same problem of flushing?

**Comment by MohamedElKamhawy** [comment]
@msftristew  Yea, I found it here: http://stackoverflow.com/questions/10019456/usage-of-sys-stdout-flush-method

**Comment by aggFTW** [comment]
I'm not worried about performance for this class. It's not the intended usage.

What did you find @MohamedElKamhawy?

**Comment by MohamedElKamhawy** [comment]
@aggFTW  We can use sys.sterr.flush as well

**Comment by aggFTW** [comment]
Please unit test that flush gets called on every `write` and `writeln` or some useful unit test. We don't want regressions. This also applies to the error scenarios. You need to find out if we have a problem there @MohamedElKamhawy.

**Comment by msftristew** [comment]
Mohamed, are you going to add a unit test for this?

**Comment by MohamedElKamhawy** [comment]
@msftristew  Yes, I'm trying that but have some errors.

File: tests/test_ipythondisplay.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+def test_stdout_flush():
+    ipython_shell = MagicMock()
+    ipython_display = IpythonDisplay()
+    ipython_display._ipython_shell = ipython_shell
+    sys.stdout = MagicMock()
+
+    ipython_display.write('Testing Stdout Flush')
+    assert sys.stdout.flush.call_count == 1
+
```
Comment (aggFTW) [inline tests/test_ipythondisplay.py]: Nit: double new line between methods per PEP specification.

File: remotespark/livyclientlib/livysession.py [hunk 5]
```diff
@@ -4,6 +4,7 @@
 import textwrap
 from time import sleep, time
 import remotespark.utils.configuration as conf
+
```
Comment (aggFTW) [inline remotespark/livyclientlib/livysession.py]: Nit: Remove line.

**Comment by msftristew** [comment]
LGTM

**Pull request merged**